cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/network/model/optimized/ConsentStatusApiModel.kt

```diff
@@ -36,45 +34,48 @@ enum class GranularState {
 
 enum class GCMStatus(val status: String) {
     GRANTED("granted"),
-    DENIED("denied"),
-}
-
-@Serializable
-data class ConsentStatusResp(
-    @SerialName("consentStatusData") val consentStatusData: ConsentStatusData?,
-    @SerialName("localState") val localState: JsonElement?
-) {
-    @Serializable
-    data class ConsentStatusData(
-        @SerialName("ccpa") val ccpa: CcpaCS?,
-        @SerialName("gdpr") val gdpr: GdprCS?,
-        @SerialName("usnat") val usnat: USNatConsentData?,
-    )
+    DENIED("denied");
 
-    override fun toString(): String {
-        return check { JsonConverter.converter.encodeToString(this) }.getOrNull()
-            ?: super.toString()
+    companion object {
+        fun firstWithStatusOrNull(status: String?) = entries.firstOrNull { it.status == status }
     }
 }
 
 @Serializable
 data class CcpaCS(
-    val applies: Boolean?,
-    @SerialName("consentedAll") val consentedAll: Boolean?,
-    @SerialName("dateCreated") val dateCreated: String?,
-    @SerialName("gpcEnabled") val gpcEnabled: Boolean?,
-    @SerialName("newUser") val newUser: Boolean?,
-    @SerialName("rejectedAll") val rejectedAll: Boolean?,
-    @SerialName("rejectedCategories") val rejectedCategories: List<String>?,
-    @SerialName("rejectedVendors") val rejectedVendors: List<String>?,
-    @SerialName("signedLspa") val signedLspa: Boolean?,
+    val applies: Boolean? = null,
+    @SerialName("consentedAll") val consentedAll: Boolean? = null,
+    @SerialName("dateCreated") val dateCreated: String? = null,
+    @SerialName("gpcEnabled") val gpcEnabled: Boolean? = null,
+    @SerialName("newUser") val newUser: Boolean? = null,
+    @SerialName("rejectedAll") val rejectedAll: Boolean? = null,
+    @SerialName("rejectedCategories") val rejectedCategories: List<String>? = emptyList(),
+    @SerialName("rejectedVendors") val rejectedVendors: List<String>? = emptyList(),
+    @SerialName("signedLspa") val signedLspa: Boolean? = null,
     @SerialName("uspstring") val uspstring: String? = null,
-    @Serializable(with = CcpaStatusSerializer::class) val status: CcpaStatus?,
-    @SerialName("GPPData") @Serializable(with = JsonMapSerializer::class) val gppData: Map<String, JsonElement>? = null,
-    @SerialName("uuid") var uuid: String?,
+    @Serializable(with = CcpaStatusSerializer::class) val status: CcpaStatus? = null,
+    @SerialName("GPPData") @Serializable(with = JsonMapSerializer::class) val gppData: Map<String, JsonElement>? = emptyMap(),
```

Comment:
If there a particular reason why we now initialising all these values? 
Main reason for me to ask is: few variables might become NOT nullable if they are already initialised ( `gppData`, `rejectedCategories`, `rejectedVendors`)

Reply:
Attributes that are a collection, it's a better practice to give them a default value (like empty collection) instead of a null value. The only reason I kept them as optionals (`?`) is because they might not be present in the responses we get further along the code. We should address that in the mobile-core in the future.